src/Tools/NodeDocumentationMarkdownGenerator/MdFileInfo.cs

```diff
@@ -0,0 +1,132 @@
+        public string NodeName { get; }
+
+        /// <summary>
+        /// Nodes namespace, this is used to generate the md file name
+        /// </summary>
+        public string NodeNamespace { get; }
+
+        /// <summary>
+        /// Full node category, as showed in the dynamo library
+        /// </summary>
+        public string FullCategory { get; }
+
+        /// <summary>
+        /// Name of markdown file
+        /// </summary>
+        public string FileName { get; }
+
+        /// <summary>
+        /// Name of external library this node belongs to
+        /// </summary>
+        public string ExternalLibraryName { get; }
+
+        public MdFileInfo(string nodeName, string nodeNamespace, string fullCategory, string externalLib, string fileName)
+        {
+            if (string.IsNullOrEmpty(nodeName))
+            {
+                throw new ArgumentException($"'{nameof(nodeName)}' cannot be null or empty.", nameof(nodeName));
+            }
+
+            if (string.IsNullOrEmpty(nodeNamespace))
+            {
+                throw new ArgumentException($"'{nameof(nodeNamespace)}' cannot be null or empty.", nameof(nodeNamespace));
+            }
+
+            if (string.IsNullOrEmpty(fullCategory))
+            {
+                throw new ArgumentException($"'{nameof(fullCategory)}' cannot be null or empty.", nameof(fullCategory));
+            }
+
+            if (string.IsNullOrEmpty(externalLib))
+            {
+                throw new ArgumentException($"'{nameof(externalLib)}' cannot be null or empty.", nameof(externalLib));
+            }
+
+            if (string.IsNullOrEmpty(fileName))
+            {
+                throw new ArgumentException($"'{nameof(fileName)}' cannot be null or empty.", nameof(fileName));
+            }
+
+            NodeName = nodeName;
+            NodeNamespace = nodeNamespace;
+            FullCategory = fullCategory;
+            ExternalLibraryName = externalLib;
+            FileName = fileName;
+        }
+
+        internal static MdFileInfo FromCustomNode(string path)
+        {
+            WorkspaceInfo header = null;
+            ILogger log = new DummyConsoleLogger();
+
+            if (DynamoUtilities.PathHelper.isValidXML(path, out XmlDocument xmlDoc, out Exception ex))
+            {
+                WorkspaceInfo.FromXmlDocument(xmlDoc, path, true, false, log, out header);
+            }
+            else if (DynamoUtilities.PathHelper.isValidJson(path, out string jsonDoc, out ex))
+            {
+                WorkspaceInfo.FromJsonDocument(jsonDoc, path, true, false, log, out header);
+            }
+            else throw ex;
+
+            if (!header.IsVisibleInDynamoLibrary) return null;
+
+            var nodeName = header.Name;
+            var nodeNamspace = header.Category;
+            var fullCategory = $"{header.Category}.{header.Name}";
+            return new MdFileInfo(nodeName, nodeNamspace, fullCategory, header.Category, nodeNamspace);
```

Comment:
why is the namespace is the same as the file name?

Reply:
Based on how we look up documentation in the doc browser i would say it should be. Do you have any examples where it wouldn't be?

Reply:
Won't it be different if the node is overloaded and there are two nodes with the same name but different arguments?

Reply:
@mjkkirschner yea you are right. This might be a little tricky though. The way we handle CN overloads in the `OpenNodeAnnotationEventArgs` is to check if the package that the node comes from has any other `LoadedCustomNodes` with the same name, if it does we extract the input from the nodes `InPorts`.

In this case we cant really get the inputs from CNs as there arent an actual node in this case we only have the json file, which as far as i can see does not tell us anything about the inputs of the node. We can look for all `InputNodes` and just use those i guess, but that means we have to serialize the cn as a `Workspace` and not a `WorkspaceInfo`.

On another note, is it actually possible to have overloaded CNs? from my testing it seems like it will only show one of them if you have two CNs with the same `Name`, i might be doing something wrong though.

Reply:
I think you can have custom nodes with the same name as long as they have different function ids